hera_cal/io.py

```diff
@@ -0,0 +1,286 @@
+        ants: ndarray containing unique antenna indices
+        freqs: ndarray containing frequency channels (Hz)
+        times: ndarray containing julian date bins of data
+        lsts: ndarray containing LST bins of data (radians)
+        pol: ndarray containing list of polarization strings
+    '''
+
+    uvd = UVData()
+    if isinstance(input_data, (tuple, list, np.ndarray)): #List loading
+        if np.all([isinstance(id, str) for id in input_data]): #List of visibility data paths
+            if format is 'miriad':
+                uvd.read_miriad(list(input_data))
+            elif format is 'uvfits':
+                 #TODO: implement this
+                raise NotImplementedError('This function has not been implemented yet.')
+            else:
+                raise NotImplementedError("Data format must be either 'miriad' or 'uvfits'.")
+        elif np.all([isinstance(id, UVData) for id in input_data]): #List of uvdata objects
+            uvd = reduce(operator.add, input_data)
+        else:
+            raise TypeError('If input is a list, it must be only strings or only UVData objects.')
+    elif isinstance(input_data, str): #single visibility data path
+        if format is 'miriad':
+            uvd.read_miriad(input_data)
+        elif format is 'uvfits':
+             #TODO: implement this
+            raise NotImplementedError('This function has not been implemented yet.')
+        else:
+            raise NotImplementedError("Data format must be either 'miriad' or 'uvfits'.")
+    elif isinstance(input_data, UVData): #single UVData object
+        uvd = input_data
+    else:
+        raise TypeError('Input must be a UVData object, a string, or a list of either.')
+
+    data, flags = {}, {}
+    # create nested dictionaries of visibilities in the data[bl][pol] format
+    for nbl, (i, j) in enumerate(map(uvd.baseline_to_antnums, np.unique(uvd.baseline_array))):
+        if (i, j) not in data:
+            data[i, j] = {}
+            flags[i, j] = {}
+        for ip, pol in enumerate(uvd.polarization_array):
+            pol = polnum2str(pol)
+            new_data = deepcopy(uvd.get_data((i, j, pol)))
+            new_flags = deepcopy(uvd.get_flags((i, j, pol)))
+            if pol not in data[(i, j)]:
+                data[(i, j)][pol] = new_data
+                flags[(i, j)][pol] = new_flags
+            else:
+                data[(i, j)][pol] = np.concatenate([data[(i, j)][pol], new_data ])
+                flags[(i, j)][pol] = np.concatenate([flags[(i, j)][pol], new_flags ])
+    
+    if pop_autos:
+        for bl in data.keys():
+            if bl[0] == bl[1]:
+                del data[bl], flags[bl]
+
+    # If we don't want nested dicts, convert to DataContainer
+    if not nested_dict:
+        data, flags = DataContainer(data), DataContainer(flags)
+
+    # get meta
+    if return_meta:
+        freqs = np.unique(uvd.freq_array)
```

Comment:
Why are you doing `unique` here? It has an odd side effect of returning an array of shape `(Nfreq,)` instead of `(1, Nfreq)`, which is fine except that you aren't doing it in `load_cal`, so there you get the latter shape.

Reply:
Good point. I'd think I prefer the shape to be (Nfreq,) (since we're not supporting multiple spectra windows), so I'll put unique in `load_cal`